posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,800 @@
+
+**Details:** wg-debugging has been spun up. We now are conducting regular status/design and triage meetings. We are working through the backlog of A-debuginfo issues. There is active participation from a number of contributors.
+
+### improving debuginfo quality
+
+<!-- https://hackmd.io/@rust-compiler-team/HJQ25sXOq -->
+
+**Progress:** The Rust project has made progress on this ambition since the start of the year
+
+**Goals:** We had no goals for this ambition planned for this year, but we made ad-hoc progress on the problem itself.
+There’s nearly an unbounded amount of effort that could be spent improving debuginfo quality but wesleywiser thinks we are making significant improvement both over the last 6 months and in the final 6 months of this year as well.
+
+**How it started:** At start of 2022, we knew a problem existed, but we did not yet have a specific goal in mind for solving the problem.
+
+**How it's going:** We would like help deciding what to do next.
+We understand the problem better than we did at the start of the year,  we have Rust contributors who have agreed to help with the units of work that we have identified,
+Furthermore, (some of) our work has reached Rust programmers. In some cases, we do not know if it has improved Rust for them; in others, what we learn of their usage is informing our plans going forward.
+Much of the work wesleywiser is aware of has landed in 1.60 or 1.61 but there are a few small pieces landing in 1.62 (current beta).
+
+**Details:**
+We've made concrete improvements/fixes to debuginfo generation.
+
+Specifically:
+- Change char type in debuginfo to DW_ATE_UTF [#89887](https://github.com/rust-lang/rust/pull/89887)
+- Fix debuginfo for pointers/references to unsized types [#93006](https://github.com/rust-lang/rust/pull/93006)
+- debuginfo: Support fat pointers to unsized tuples. [#94050](https://github.com/rust-lang/rust/pull/94050)
+- debuginfo: Fix bug in type name generation for dyn types with associated types but no other generic arguments. [#94810](https://github.com/rust-lang/rust/pull/94810)
+- async: Give predictable name to binding generated from .await expressions. [#95011](https://github.com/rust-lang/rust/pull/95011)
+- debuginfo: Fix debuginfo for Box&lt;T&gt; where T is unsized. [#95270](https://github.com/rust-lang/rust/pull/95270)
+- debuginfo: Emit ZST struct debuginfo for unit type when CPP-like debuginfo is enabled [#96316](https://github.com/rust-lang/rust/pull/96316)
+
+Surprises:
+Debuginfo just doesn't have enough test coverage, but that isn't particularly surprising.
+
+**Regarding prioritization and focus:** debugging in general is a top priority for mw & wesleywiser’s
+team.
+
+### supporting split debuginfo
+
+<!-- https://hackmd.io/@rust-compiler-team/ByXfjiXu5 -->
+
+**Progress:** The Rust project has not made any progress on this ambition since the start of the year (pnkfelix: but the free form answer somewhat contradicts this)
```

Comment:
Nit

```suggestion
**Progress:** The Rust project has not made any progress on this ambition since the start of the year (@pnkfelix: but the free form answer somewhat contradicts this)
```